app/oxalis/security/OpenIdConnectClient.scala

```diff
@@ -0,0 +1,93 @@
+package oxalis.security
+
+import com.scalableminds.util.tools.Fox
+import com.scalableminds.util.tools.Fox.{jsResult2Fox, try2Fox}
+import com.scalableminds.webknossos.datastore.rpc.RPC
+import play.api.libs.json.{JsObject, Json, OFormat}
+import pdi.jwt.{JwtJson, JwtOptions}
+import play.api.libs.ws._
+import utils.WkConf
+
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import javax.inject.Inject
+import scala.concurrent.ExecutionContext
+
+class OpenIDConnectClient @Inject()(rpc: RPC, conf: WkConf)(implicit executionContext: ExecutionContext) {
+
+  /*
+   Build redirect URL to redirect to OIDC provider for auth request (https://openid.net/specs/openid-connect-core-1_0.html#AuthRequest)
+   */
+  def redirectUrl(openIdClient: OpenIdConnectConfig, redirectUrl: String): Fox[String] =
```

Comment:
```suggestion
  def getRedirectUrl: Fox[String] =
```

I’d add the `get` because `redirect` also reads a bit like a verb.

I’d say it is not necessary to pass the parameters in from the caller, but rather have lazy vals for the client itself, I thinkt it already has everything it needs to construct them (mostly the injected WkConf).
Maybe the OpenIdConnectConfig case class is not needed at all, but the fields can be read from the passed WkConf where needed?